analyzers/src/SonarAnalyzer.CSharp/Rules/NonDerivedPrivateClassesShouldBeSealed.cs

```diff
@@ -42,7 +43,9 @@ public sealed class NonDerivedPrivateClassesShouldBeSealed : SonarDiagnosticAnal
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
             {
                 var typeDeclarationSyntax = (TypeDeclarationSyntax)c.Node;
-                if (IsPrivateButNotSealedType(typeDeclarationSyntax) && !HasVirtualMembers(typeDeclarationSyntax))
+                if (IsPrivateButNotSealedType(typeDeclarationSyntax)
+                    && !HasVirtualMembers(typeDeclarationSyntax)
+                    && !c.IsRedundantPositionalRecordContext())
```

Comment:
_Performance:_ The redundant stuff should go first. The other two conditions are expensive compared to it.